site/content/contribute/more-info/server/tests.md

```diff
@@ -21,10 +21,79 @@ A flaky test is one that exhibits both passing and failing results when run mult
    - Copy the test failure message into the JIRA ticket description
    - Add the `flaky-test` and `triage-global` labels
    - Create a PR to skip the test by adding:
+
      ```go
      t.Skip("https://mattermost.atlassian.net/browse/MM-XXXXX")
      ```
+
      where MM-XXXXX is your JIRA ticket number
    - Link the JIRA ticket in the skip message for tracking
 
 This process helps us track and systematically address flaky tests while preventing them from blocking development work.
+
+## Writing Parallel Tests
+
+Leveraging parallel tests can drastically reduce execution time for entire test packages, such as `api4` and `app`, which are notably heavy with hundreds of tests. However, careful implementation is essential to ensure reliability and prevent flakiness. Follow these guidelines when writing parallel tests:
+
+### Enabling Parallel Tests
+
+Ensure parallel tests run behind an explicit environment flag to control execution:
+
+```go
+// In api4 and app packages, check mainHelper.Options.RunParallel
+if mainHelper.Options.RunParallel {
+    t.Parallel()
+}
+
+// In sqlstore package, a global enableFullyParallelTests variable is used instead
+if enableFullyParallelTests {
+    t.Parallel()
+}
+```
+
+To enable parallel execution, you should set the `ENABLE_FULLY_PARALLEL_TESTS` environment variable. Example:
+
+```bash
+ENABLE_FULLY_PARALLEL_TESTS=true go test -v ./api4/...
```

Comment:
Can we prefix this with `MM_` to keep the namespacing sane?

Reply:
I'm thinking of examples like `MM_DEBUG` for plugins, but happy to also change perspective. Who is the target audience for confusion?

Reply:
This is meant as a developer setting, but mostly for CI. We do use the `CI_` prefix for a few things, but I also wanted to give the opportunity to run the tests on a local environment now that it doesn't take hours to complete them so I ended up prefix-less :p

I am totally happy to add the `MM_` prefix, just sharing some context.

Reply:
I'm 0/5... if you can write down your score I'll happily defer :P